include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -366,15 +355,14 @@ namespace internal
 
 
 template <int structdim, int dim, int spacedim>
-inline dealii::internal::TriangulationImplementation::TriaObjects<
-  dealii::internal::TriangulationImplementation::TriaObject<structdim>> &
+inline dealii::internal::TriangulationImplementation::TriaObjects &
 TriaAccessorBase<structdim, dim, spacedim>::objects() const
 {
   if (structdim != dim)
     // get sub-objects. note that the
     // current class is only used for
     // objects that are *not* cells
-    return *dealii::internal::TriaAccessorBaseImplementation::get_objects(
+    return *dealii::internal::TriaAccessorBaseImplementation::get_objects<dim>(
       this->tria->faces.get(), std::integral_constant<int, structdim>());
```

Comment:
Is this complicated function call setup still needed? Given that we do not need to distinguish types via their template argument any more, can't we just switch to faces/lines as appropriate?

Reply:
Indeed:

```cpp
template <int structdim, int dim, int spacedim>
inline dealii::internal::TriangulationImplementation::TriaObjects &
TriaAccessorBase<structdim, dim, spacedim>::objects() const
{
  if (structdim == dim)
    return this->tria->levels[this->present_level]->cells;


  if (structdim == 1 && dim > 1)
    return this->tria->faces->lines;


  if (structdim == 2 && dim > 2)
    return this->tria->faces->quads;


  Assert(false, ExcInternalError());


  return *static_cast<
    dealii::internal::TriangulationImplementation::TriaObjects *>(nullptr);
}
```

Now, the code is readable!